fhevm/sgx_arithmetic_test.go

```diff
@@ -0,0 +1,145 @@
+package fhevm
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/fhevm/tfhe"
+	"github.com/zama-ai/fhevm-go/sgx"
+)
+
+func TestSgxAddRun(t *testing.T) {
+	op := func(lhs, rhs uint64) uint64 {
+		return lhs + rhs
+	}
+	signature := "sgxAdd(uint256,uint256,bytes1)"
+
+	testcases := []struct {
+		typ tfhe.FheUintType
+		lhs uint64
+		rhs uint64
+	}{
+		{tfhe.FheUint4, 2, 1},
+		{tfhe.FheUint8, 2, 1},
+		{tfhe.FheUint16, 4283, 1337},
+		{tfhe.FheUint32, 1333337, 1337},
+		{tfhe.FheUint64, 13333377777777777, 133377777777},
+	}
+	for _, tc := range testcases {
+		t.Run(fmt.Sprintf("sgxAdd with %s", tc.typ), func(t *testing.T) {
+			sgxArithmeticHelper(t, tc.typ, tc.lhs, tc.rhs, op, signature)
+		})
+	}
+}
+
+func TestSgxSubRun(t *testing.T) {
+	op := func(lhs, rhs uint64) uint64 {
+		return lhs - rhs
+	}
+	signature := "sgxSub(uint256,uint256,bytes1)"
+
+	testcases := []struct {
+		typ tfhe.FheUintType
+		lhs uint64
+		rhs uint64
+	}{
+		{tfhe.FheUint4, 2, 1},
+		{tfhe.FheUint8, 2, 1},
+		{tfhe.FheUint16, 4283, 1337},
+		{tfhe.FheUint32, 1333337, 1337},
+		{tfhe.FheUint64, 13333377777777777, 133377777777},
+	}
+	for _, tc := range testcases {
+		t.Run(fmt.Sprintf("sgxAdd with %s", tc.typ), func(t *testing.T) {
```

Comment:
We have to change from `sgxAdd` to `sgxSub`?

Reply:
Done